astroquery/desi/core.py

```diff
@@ -0,0 +1,77 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+import urllib.error
+import requests
+
+import pyvo as vo
+import numpy as np
+
+from astroquery.exceptions import NoResultsWarning
+from astroquery.query import BaseQuery
+from astroquery.utils import commons, async_to_sync
+
+from . import conf
+
+
+__all__ = ['DESILegacySurvey', 'DESILegacySurveyClass']
+
+
+@async_to_sync
+class DESILegacySurveyClass(BaseQuery):
+    URL = conf.server
+    TIMEOUT = conf.timeout
+
+    def query_region(self, coordinates, radius, data_release=9):
+        """
+        Queries a region around the specified coordinates.
+
+        Parameters
+        ----------
+        coordinates : str or `astropy.coordinates`.
+            coordinates around which to query
+        radius : str or `astropy.units.Quantity`.
+            the radius of the cone search
+
+        Returns
+        -------
+        response : `astropy.table.Table`
+        """
+
+        url = 'https://datalab.noirlab.edu/tap'
+        tap_service = vo.dal.TAPService(url)
+        qstr = "SELECT all * FROM ls_dr" + str(data_release) + ".tractor WHERE dec>" + str(coordinates.dec.deg - radius.deg) + " and dec<" + str(
+            coordinates.dec.deg + radius.deg) + " and ra>" + str(coordinates.ra.deg - radius.deg / np.cos(coordinates.dec.deg * np.pi / 180.)) + " and ra<" + str(
+            coordinates.ra.deg + radius.deg / np.cos(coordinates.dec.deg * np.pi / 180))
+
+        tap_result = tap_service.run_sync(qstr)
+        tap_result = tap_result.to_table()
+        # filter out duplicated lines from the table
+        mask = tap_result['type'] != 'D'
+        filtered_table = tap_result[mask]
+
+        return filtered_table
+
+    def get_images(self, position, data_release=9, pixels=None, radius=None, show_progress=True, image_band='g'):
```

Comment:
Make optionals kwarg only. Please do this for the rest of the methods, too, I won't comment on it elsewhere.

```suggestion
    def get_images(self, position, *, data_release=9, pixels=None, radius=None, show_progress=True, image_band='g'):
```